src/utils/sshUtils.ts

```diff
@@ -6,25 +6,30 @@
 import * as fse from 'fs-extra';
 import * as os from "os";
 import { join } from 'path';
+import { callWithMaskHandling } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { VirtualMachineTreeItem } from '../tree/VirtualMachineTreeItem';
 import { cpUtils } from "./cpUtils";
 
 export const sshFsPath: string = join(os.homedir(), '.ssh');
 
-export async function getSshKey(vmName: string): Promise<string> {
-    const sshKeyName: string = `azure_${vmName}_rsa`;
-    const sshKeyPath: string = join(sshFsPath, sshKeyName);
+export async function getSshKey(vmName: string, passphrase: string): Promise<string> {
+    return await callWithMaskHandling(
+        async () => {
+            const sshKeyName: string = `azure_${vmName}_rsa`;
+            const sshKeyPath: string = join(sshFsPath, sshKeyName);
 
-    if (!await fse.pathExists(`${sshKeyPath}.pub`)) {
-        // if the SSH key doesn't exist, create it
-        const sshKeygenCmd: string = `ssh-keygen -t rsa -b 2048 -f ${sshKeyPath} -N ""`;
-        ext.outputChannel.appendLog(localize('generatingKey', `Generating public/private rsa key pair with command "${0}".`, sshKeygenCmd));
-        await cpUtils.executeCommand(undefined, undefined, sshKeygenCmd);
-    }
+            if (!await fse.pathExists(`${sshKeyPath}.pub`)) {
+                // if the SSH key doesn't exist, create it
+                const sshKeygenCmd: string = `ssh-keygen -t rsa -b 4096 -f ${sshKeyPath} -N "${passphrase}"`;
```

Comment:
Instead of always using double quotes for passphrase, use `wrapArgInQuotes` like here otherwise passphrases with `$` won't work on mac/linux: https://github.com/microsoft/vscode-azurefunctions/blob/master/src/utils/cpUtils.ts#L89

Also might be easier to pass the args in like this:
```
await cpUtils.executeCommand(undefined, undefined, 'ssh-keygen', '-t', 'rsa', '-b', '4096', '-f', cpUtils.wrapArgInQuotes(sshKeyPath), '-N', cpUtils.wrapArgInQuotes(passphrase));
```